src/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQEnums.cpp

```diff
@@ -364,54 +363,81 @@ std::string typeName(UDQVarType var_type) {
+        {UDAControl::WCONPROD_ORAT, UDAKeyword::WCONPROD},
+        {UDAControl::WCONPROD_GRAT, UDAKeyword::WCONPROD},
+        {UDAControl::WCONPROD_WRAT, UDAKeyword::WCONPROD},
+        {UDAControl::WCONPROD_LRAT, UDAKeyword::WCONPROD},
+        {UDAControl::WCONPROD_RESV, UDAKeyword::WCONPROD},
+        {UDAControl::WCONPROD_BHP,  UDAKeyword::WCONPROD},
+        {UDAControl::WCONPROD_THP,  UDAKeyword::WCONPROD},
+        {UDAControl::WCONINJE_RATE, UDAKeyword::WCONINJE},
+        {UDAControl::WCONINJE_RESV, UDAKeyword::WCONINJE},
+        {UDAControl::WCONINJE_BHP,  UDAKeyword::WCONINJE},
+        {UDAControl::WCONINJE_THP,  UDAKeyword::WCONINJE},
+        {UDAControl::GCONPROD_OIL_TARGET,    UDAKeyword::GCONPROD},
+        {UDAControl::GCONPROD_WATER_TARGET,  UDAKeyword::GCONPROD},
+        {UDAControl::GCONPROD_GAS_TARGET,    UDAKeyword::GCONPROD},
+        {UDAControl::GCONPROD_LIQUID_TARGET, UDAKeyword::GCONPROD},
+        {UDAControl::GCONINJE_SURFACE_MAX_RATE,      UDAKeyword::GCONINJE},
+        {UDAControl::GCONINJE_RESV_MAX_RATE,         UDAKeyword::GCONINJE},
+        {UDAControl::GCONINJE_TARGET_REINJ_FRACTION, UDAKeyword::GCONINJE},
+        {UDAControl::GCONINJE_TARGET_VOID_FRACTION,  UDAKeyword::GCONINJE},
+    };
+
+    return lookup_control_map_value(c2k, control);
 }
 
-
-int uadCode(UDAControl control) {
-    const std::map<UDAControl, int> c2uad = {{UDAControl::WCONPROD_ORAT,  300004},
-                                             {UDAControl::WCONPROD_GRAT,  500004},
-                                             {UDAControl::WCONPROD_WRAT,  400004},
-                                             {UDAControl::WCONPROD_LRAT,  600004},
-                                             {UDAControl::WCONPROD_RESV,  999999},
-                                             {UDAControl::WCONPROD_BHP,   999999},
-                                             {UDAControl::WCONPROD_THP,   999999},
-                                             {UDAControl::WCONINJE_RATE,  400003},
-                                             {UDAControl::WCONINJE_RESV,  500003},
-                                             {UDAControl::WCONINJE_BHP,   999999},
-                                             {UDAControl::WCONINJE_THP,   999999},
-                                             {UDAControl::GCONPROD_OIL_TARGET,    200019},
-                                             {UDAControl::GCONPROD_WATER_TARGET,  300019},
-                                             {UDAControl::GCONPROD_GAS_TARGET,    400019},
-                                             {UDAControl::GCONPROD_LIQUID_TARGET, 500019}};
-
-    auto it = c2uad.find(control);
-    if (it != c2uad.end())
-      return it->second;
-
-    throw std::logic_error("Unrecognized enum type - internal error");
-}
-}
+int uadCode(UDAControl control)
+{
+    static const auto c2uad = std::map<UDAControl, int> {
+        {UDAControl::WCONPROD_ORAT,  300'004},
```

Comment:
What is this `300'004` syntax? If it is a thousand separator I would prefer that it was removed.

Reply:
> If it is a thousand separator I would prefer that it was removed

It is indeed a separator, but it's not specific to groups of three digits.  Compilers are required to ignore single quote characters in this specific context.  The syntax was introduced in C++14 as a means of making source code [integer literals](https://en.cppreference.com/w/cpp/language/integer_literal) with "many" digits easier to read.  Although I would personally have preferred Perl's underscore syntax (e.g., `1_000_000` for the literal representation of one million), I think this a decent enough compromise.

In any case, I've pushed an update which removes it.